pkgs/by-name/vm/vmaware/package.nix

```diff
@@ -0,0 +1,67 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  ninja,
+  dmidecode,
+  makeWrapper,
+  versionCheckHook,
+  nix-update-script,
+}:
+
+let
+  isPlatformSupported = drv: builtins.elem stdenv.hostPlatform.system drv.meta.platforms;
+  runtimeDeps = builtins.filter isPlatformSupported [
+    dmidecode
+  ];
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "vmaware";
+  version = "2.1";
+  src = fetchFromGitHub {
+    owner = "kernelwernel";
+    repo = "VMAware";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-lnDxgSvElm8swSxVm+s7S9wz1Fy9Jb/Z+sLngowtnA0=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    ninja
+    makeWrapper
+  ];
+
+  patchPhase = ''
+    runHook prePatch
+
+    substituteInPlace CMakeLists.txt --replace-fail "/usr/local/bin" "$out/bin"
+    substituteInPlace CMakeLists.txt --replace-fail "/usr/include" "$out/include"
+
+    runHook postPatch
+  '';
+
+  postFixup = ''
+    wrapProgram $out/bin/vmaware \
+      --prefix PATH : "${lib.makeBinPath runtimeDeps}"
+  '';
+
+  nativeInstallCheckInputs = [ versionCheckHook ];
+  versionCheckProgramArg = [ "--version" ];
+  doInstallCheck = true;
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    description = "VM detection library and tool";
+    homepage = "https://github.com/kernelwernel/VMAware";
+    changelog = "https://github.com/kernelwernel/VMAware/releases/tag/v${finalAttrs.version}";
+    license = with lib.licenses; [
+      gpl3
+      mit
+    ];
```

Comment:
The licensing may be a bit too granular to represent simply as "one of MIT or GPL3."

The source repository describes the MIT licensed version as having fewer capabilities, but the Nixpkgs version will not be conditionally built with different capabilities depending on the license. Not sure the best option, but perhaps one option is to only list it as GPL3 here and add a code comment about why MIT was omitted.

Reply:
I apologize, I somehow missed that the list form of licenses merely indicates _parts_ of the package are under each listed license. So that's appropriate here.